### PR TITLE
docs(rust): document the layout of files for a node

### PR DESCRIPTION
**Pull request by etorreborre**
This PR provides an overview of all the data stored to disk when using the `ockam_api` crate with a `NodeManager`.

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/README.md [hunk 1]
```diff
@@ -11,6 +11,123 @@ and trustfully with cloud services and other devices.
...
+      ├─ vault1.lmdb
+      ├─ vault1.lmdb-lock
+      ├─ vault2.lmdb
+      ├─ vault2.lmdb-lock
+      └─ ...
+```
+## `credentials`
+
+Each file stored under the `credentials` directory contains the credential for a given identity.
+Those files are created with the `ockam credential store` command. They are then read during the creation of
```
Comment (hairyhum) [inline implementations/rust/ockam/ockam_api/README.md:66]: In the use-case examples we retrieve credentials via `ockam project authenticate/enroll` using enrollment tokens. Are those credentials also saved here?

**Review by caoakleyii — DISMISSED** [review]
Lgtm

File: implementations/rust/ockam/ockam_api/README.md [hunk 2]
```diff
@@ -11,6 +11,123 @@ and trustfully with cloud services and other devices.
...
+## `defaults`
+
+This directory contains symlinks to other files or directories in order to specify which node,
+identity, credential or vault must be considered as a default when running a command expecting those
+inputs
+
+## `identities`
+
+This directory contains one file per identity and a data directory. An identity file is created
+with the `ockam identity create` command or created by default for some commands (in that case the
```
Comment (hairyhum) [inline implementations/rust/ockam/ockam_api/README.md:78]: Would identities created by the identity service be also stored here? If not are they stored anywhere?

File: implementations/rust/ockam/ockam_api/README.md [hunk 3]
```diff
@@ -11,6 +11,123 @@ and trustfully with cloud services and other devices.
...
+`defaults/identity` symlink points to that identity). The identity file contains:
+
+- the identity identifier
+- the enrollment status for that identity
+
+The `data` directory contains a LMDB database with other information about identities:
```
Comment (hairyhum) [inline implementations/rust/ockam/ockam_api/README.md:84]: Is that information shared by all nodes? Which nodes read or write this LMDBs?

File: implementations/rust/ockam/ockam_api/README.md [hunk 4]
```diff
@@ -11,6 +11,123 @@ and trustfully with cloud services and other devices.
...
+Note: for each `.lmdb` file there is a corresponding `lmdb-lock` file which is used to control
+the exclusive access to the LMDB database even if several OS processes are trying to modify it.
+
+## `nodes`
+
+This directory contains:
+
+ - symlinks to default values for the node: identity and vault
+ - a database for ABAC policies
+ - a setup file containing some configuration information for the node (is it an authority node?, what is the TCP listener address?,...)
```
Comment (hairyhum) [inline implementations/rust/ockam/ockam_api/README.md]: Is this file generated by the node? Who writes and reads this file?

File: implementations/rust/ockam/ockam_api/README.md [hunk 5]
```diff
@@ -11,6 +11,123 @@ and trustfully with cloud services and other devices.
...
+ - log files: for system errors and system outputs. The stdout.log file is where almost all the node logs are written
+ - a version number for the configuration
+
+## `projects`
+
+This directory contains a list of files, one per project that was created, either the default project
```
Comment (hairyhum) [inline implementations/rust/ockam/ockam_api/README.md:113]: If we start a node in the environment which has project configurations, does the node have access to these configurations?

**Review by hairyhum — CHANGES_REQUESTED** [review]
I think this also needs some information about which process, command or node creates and reads which files to understand the lifecycle of the files.

File: implementations/rust/ockam/ockam_api/README.md (hunk 1 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/README.md:66]: No they are only stored when we do `ockam credential store`

File: implementations/rust/ockam/ockam_api/README.md (hunk 3 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/README.md:84]: This database is shared by all the nodes which are started using the same `OCKAM_HOME`. All nodes can read and write this database, for example when they create a secure channel and store the other party identity.

File: implementations/rust/ockam/ockam_api/README.md (hunk 2 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/README.md:78]: The identity service would store an identity directly in the shared database and no file would be created.

File: implementations/rust/ockam/ockam_api/README.md (hunk 4 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/README.md]: This information is created when we create the node (regular or authority). It is read again if we restart the node.

File: implementations/rust/ockam/ockam_api/README.md (hunk 5 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/README.md:113]: When you start a node you either specify nothing and that node uses the default project or you specify one of the project configurations.

**Review by hairyhum — APPROVED** [review]

**Pull request merged**